docs/src/apalache/config.md

```diff
@@ -1,23 +1,72 @@
-  - `profiling` (Bool): This flag governs the creation of `profile-rules.txt` used in [profiling](profiling.md). The file is only created if `profiling` is set to `True`.  Setting `profiling` to `False` is incompatible with the `--smtprof` flag.
-  The default is `False`.
-  - `write-intermediate` (Bool): This Boolean flag governs the creation of intermediate outputs. If set to `True`, apalache will produce an `intermediate` subdirectory in the run directory and output the state of the module after each pass.
-  The default is `False`.
-
-
-Example of a valid `apalache-global-config.yml`:
-```
-out-dir: '~/apalache-out'
-profiling: True
-write-intermediate: True
+# Apalache configuration
+
+Apalache supports configuration of some parameters governing its behavior.
+
+Application configuration is loaded from the following four sources:
+
+1. Command line arguments
+2. Environment variables
+3. A local configuration file
+4. The global configuration file
+
+The order of precedence of the sources follows their numbering: i.e., and any
+configuration set in an earlier numbered source overrides a configuration set in
+a later numbered source.
+
+## Command line arguments and environment variables
+
+To view the available command line arguments, run Apalache with the `--help`
+flag and consult the section on [Running the Too](./running.md) for more
+details.
+
+*Some* parameters configurable the command are also configurable via environment
+variables. These parameters are noted in the CLI's inline help. If a parameter
+is configured both through a CLI argument and an environment variable, then the
+CLI argument always takes precedence. 
+
+## Configuration files
+
+### File format and supported parameters
+
+Local configuration files support JSON and the JSON superset
+[HOCON](https://github.com/lightbend/config/blob/master/HOCON.md).
+
+Here's an example of a valid configuration for the currently 
+supported parameters, along with their default values: 
+
+```yaml
+# Directory in which to write all log files and records of each run
+out-dir: "${PWD}/_apalache-out"
+
+# Whether or not to write additional files, that report on intermediate
+# processing steps
+write-intermediate: false
+
+# Whether or not to write the SMT profiling file
+profiling: false
+
+# Fixed directory into which generated files are written (absent by default)
+# run-dir: ~/my-run-dir
 ```
+
+A `~` found at the beginning of a file path will expanded into the value set for
+the user's home directory.
+
+Details on the effect of these parameters can be found in [Running the
+Too](./running.md).
+
+### Local configuration file
+
+You can specify a local configuration file explicitly via the `config-file`
+command line argument. If this is not provided, then Apalache will look for the
+nearest `.apalache.cfg` file, beginning in the current working directory and
+searching up through its parents.
```

Comment:
I wonder if crawling up the directory tree makes sense. As a user, I know I'd hate for the tool to do unexpected things because there's a `.cfg` file 2 folders above.

Reply:
This is very common behavior. Tools you already use follow this behavior. E.g., git. Git looks for the closest `.git`, descending up parents, and uses the data and configuration from that to determine its behavior when you invoke it.

Basically creating a `.apalache.cfg` file in a directory will make all subdirectories part of a "project" that share the same config. For another example of this behavior, see the `sbt` documentation, where the scope of a project is determined by finding the closest parent directory containing a `build.sbt` file: https://www.scala-sbt.org/1.x/docs/Directories.html#Base+directory

Reply:
I suppose, but from my experience TLA+ repositories are far less branching and project-based than code repositories (typically, it's 1-4 specs in a single folder). I'm not opposed to this behavior per se, just that my assumption as a user would be that configs are either in the same folder or global. It's fine, as long as it's clearly documented, like in the paragraph above.

Reply:
If it ends up causing any confusion, we can remove this behavior. But I expect it might be helpful for many of our emerging use cases, where we are integrating the specs alongside the implementation, and as a result need to cope with all the complicated folder structures of a standard project.